tests/base/data_out_base_vtu.cc

```diff
@@ -52,32 +79,25 @@ void check(DataOutBase::VtkFlags flags,
 template <int dim, int spacedim>
 void check_all(std::ostream &log)
 {
-#if SEPARATE_FILES == 0
-  std::ostream &out = log;
-#endif
-
-  char name[100];
   DataOutBase::VtkFlags flags;
-  if (true)
-    {
-      sprintf(name, "%d%d.vtu", dim, spacedim);
-#if SEPARATE_FILES==1
-      std::ofstream out(name);
-#else
-      out << "==============================\n"
-          << name
-          << "\n==============================\n";
-#endif
-      check<dim,spacedim>(flags, out);
-    }
+  log << "==============================" << std::endl
+      << dim << spacedim << ".vtu" << std::endl
+      << "==============================" << std::endl;
+  check<dim,spacedim>(flags, log);
 }
 
 int main()
 {
+  std::stringstream ss;
+  check_all<1,1>(ss);
+  check_all<1,2>(ss);
+  check_all<2,2>(ss);
+  check_all<2,3>(ss);
+  check_all<3,3>(ss);
+
+  std::stringstream out;
+  filter_out_xml_key(ss, "DataArray",out);
```

Comment:
if you make the last argument of the function a `std::ostream`, then you don't need the `out` object here but can just pass `deallog.get_file_stream()`.